h2o-core/src/main/java/water/util/ArrayUtils.java

```diff
@@ -1781,7 +2006,7 @@ public FrameToArray(int startCol, int endCol, long rowNum, double[][] frameConte
         _frameContent = MemoryManager.malloc8d(_rowNum, colNum);
       } else {  // make sure we are passed the correct size 2-D double array
         assert (_rowNum == frameContent.length && frameContent[0].length == colNum);
-        for (int index = 0; index < colNum; index++) { // zero fill use array
+        for (int index = 0; index < _rowNum; index++) { // zero fill use array
```

Comment:
Okay, you scared me enough that I have writeen three tests to make sure my change is correct.  I will fix tests that failed because of my change when they show up.  However, I need to first make sure that the implementation is correct.

Reply:
Sure, I just need to warn you. :)